pipeline/snapcraft/snapcraft.go

```diff
@@ -0,0 +1,152 @@
+// Package snapcraft implements the Pipe interface providing Snapcraft bindings.
+package snapcraft
+
+import (
+	"errors"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/goreleaser/goreleaser/context"
+	"golang.org/x/sync/errgroup"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// ErrNoSnapcraft is shown when snapcraft cannot be found in $PATH
+var ErrNoSnapcraft = errors.New("snapcraft not present in $PATH")
+
+// SnapcraftMetadata to generate the snap package
+type SnapcraftMetadata struct {
+	Name          string
+	Version       string
+	Summary       string
+	Description   string
+	Grade         string `yaml:",omitempty"`
+	Confinement   string `yaml:",omitempty"`
+	Architectures []string
+	Apps          map[string]AppsMetadata
+	Parts         map[string]PartsMetadata
+}
+
+// AppsMetadata for the binaries that will be in the snap package
+type AppsMetadata struct {
+	Command string
```

Comment:
There is one command per app, but there can be multiple apps per snap.

So I added a loop in there that turns each binary into an app.